kube-runtime/src/events.rs

```diff
@@ -138,22 +182,26 @@ impl From<&str> for Reporter {
 ///     instance: std::env::var("CONTROLLER_POD_NAME").ok(),
 /// };
 ///
+/// let recorder = Recorder::new(client, reporter);
+///
 /// // references can be made manually using `ObjectMeta` and `ApiResource`/`Resource` info
 /// let reference = ObjectReference {
 ///     // [...]
 ///     ..Default::default()
 /// };
 /// // or for k8s-openapi / kube-derive types, use Resource::object_ref:
 /// // let reference = myobject.object_ref();
-///
-/// let recorder = Recorder::new(client, reporter, reference);
-/// recorder.publish(Event {
-///     action: "Scheduling".into(),
-///     reason: "Pulling".into(),
-///     note: Some("Pulling image `nginx`".into()),
-///     type_: EventType::Normal,
-///     secondary: None,
-/// }).await?;
+/// recorder
+///     .publish(
+///         Event {
+///             action: "Scheduling".into(),
+///             reason: "Pulling".into(),
+///             note: Some("Pulling image `nginx`".into()),
+///             type_: EventType::Normal,
+///             secondary: None,
+///         },
+///         &reference,
+///     ).await?;
```

Comment:
i had originally "curried" the object reference in the `Recorder` creation because it made the publish call look more to the point, since it's usually created in the reconciler for the purpose of sending events about a particular object.

we can probably still cache the reference in the recorder to avoid making this breaking change, but maybe it's better to lift the reference into our `Event` struct if we want to make `.publish` a thing that works for multiple objects. do you have any preferences?

Reply:
Here we are replicating the behavior of client-go. I think that the Recorder should be initialized at the controller level to keep the cache on memory and we have to send the object reference for each publish event.

Reply:
agree, i misunderstood the way the cache would have to work. resolving this based on other thread.